pkg/stacks/unpack_test.go

```diff
@@ -673,167 +645,210 @@ status:
-    plural: secondcousins
-    singular: secondcousin
-  scope: Namespaced
-  subresources:
-    scale:
-      specReplicasPath: ""
-      statusReplicasPath: ""
-    status: {}
-  version: v1alpha1
-status:
-  acceptedNames:
-    kind: ""
-    plural: ""
-  conditions: null
-  storedVersions: null
-
----
-apiVersion: apiextensions.k8s.io/v1beta1
-kind: CustomResourceDefinition
-metadata:
-  annotations:
-    stacks.crossplane.io/group-category: Group Category
-    stacks.crossplane.io/group-overview: Group Overview
-    stacks.crossplane.io/group-overview-short: Group Short Overview
-    stacks.crossplane.io/group-readme: Group Readme
-    stacks.crossplane.io/group-title: Group Title
-    stacks.crossplane.io/icon-data-uri: data:image/svg+xml;base64,c2luZ2xlLXJlc291cmNlLW1vY2staWNvbi1kYXRhLXN2Zw==
-    stacks.crossplane.io/resource-category: Resource Category
-    stacks.crossplane.io/resource-overview: Resource Overview
-    stacks.crossplane.io/resource-overview-short: Resource Short Overview
-    stacks.crossplane.io/resource-readme: Resource Readme
-    stacks.crossplane.io/resource-title: Resource Title
-    stacks.crossplane.io/resource-title-plural: Resources Title
-    stacks.crossplane.io/stack-title: Sample Crossplane Stack
-    stacks.crossplane.io/ui-schema: |-
-      version: 0.5
-      configSections:
-      - title: group Title
-        description: group Description
-      ---
-      version: 0.5
-      configSections:
-      - title: sibling Title
-        description: sibling Description
-      ---
-      version: 0.5
-      configSections:
-      - title: kind Title
-        description: kind Description
-  creationTimestamp: null
-  labels:
-    app.kubernetes.io/managed-by: stack-manager
-    crossplane.io/scope: environment
-  name: mytypes.samples.upbound.io
+  selector:
+    matchLabels:
+      core.crossplane.io/name: "crossplane-sample-stack"
+  replicas: 1
+  template:
+    metadata:
+      name: sample-stack-controller
+      labels:
+        core.crossplane.io/name: "crossplane-sample-stack"
+    spec:
+      containers:
+      - name: sample-stack-controller
+        %s
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+`
+
+	if image != "" {
+		return fmt.Sprintf(tmpl, fmt.Sprintf("image: %s", image))
+
+	}
+
+	return fmt.Sprintf(tmpl, "")
+}
+
+func simpleBehaviorFile(sourceImage string) string {
+	tmpl := `
+crd:
+  kind: SampleClaim
+  apiVersion: samples.stacks.crossplane.io/v1alpha1
+engine:
+  type: helm2
+reconcile:
+  path: 'resources'
+source:
+  %s
+  path: /path
+`
+
+	if sourceImage != "" {
+		return fmt.Sprintf(tmpl, fmt.Sprintf("image: %s", sourceImage))
+
+	}
+
+	return fmt.Sprintf(tmpl, "")
+}
+
+func simpleAppFile(permissionScope, packageType string, includeVersion bool) string {
```

Comment:
i like this change to have functions to generate the largely common output and inject in the variables, so we aren't duplicating huge chunks of yaml everywhere.